synapseclient/core/download/download_functions.py

```diff
@@ -743,7 +743,10 @@ def _ftp_report_hook(
             )
 
             try:
-                url_has_expiration = "Expires" in urllib_urlparse.urlparse(url).query
+                url_query = urllib_urlparse.urlparse(url).query
+                url_has_expiration = (
+                    "Expires" in url_query and "X-Amz-Expires" in url_query
```

Comment:
Nit: is there a reason it isn't just `X-Amz-Expires` since it's an "and" statement?